tarodb/sqlite/queries/assets.sql

```diff
@@ -14,10 +14,12 @@ INSERT INTO asset_minting_batches (
 ) VALUES (0, ?, ?);
 
 -- name: FetchMintingBatchesByInverseState :many
-SELECT *
+SELECT batches.*, keys.*, txs.height_hint
 FROM asset_minting_batches batches
 JOIN internal_keys keys
     ON batches.batch_id = keys.key_id
+LEFT JOIN chain_txns txs
+    ON batches.genesis_id = txs.txn_id
```

Comment:
This isn't the right field to join on. 

The minting batch itself stores a raw PSBT (could change tbh, was a bit earlier before we started to solidify the usage of `chain_txns`). So I think you actually want to store to either store the height hint there, or also insert a `chain_txns` entry for it (would be duplicating tho fwiw).

Reply:
You're right, we should be joining on `anchor_tx_id`. The reason the test didn't catch this was that all ids start at 0 and gets incremented every new batch. I fixed this and added a test to exercise it correctly: https://github.com/lightninglabs/taro/pull/171/commits/a402e18341abb3277f3514bec9e1f1d4a557ba9a

Now we join on (I'm no SQL wiz, let me know if there is a simpler way to achieve it):
```SQL
LEFT JOIN chain_txns txs
    ON batches.genesis_id in (SELECT genesis_id FROM genesis_points WHERE anchor_tx_id = txs.txn_id)
```

Here is my thinking:
When inserting into `chain_txns` (`UpsertChainTx`), a `txn_id` is given to the transaction.

The `asset_minting_batches.genesis_id` references `genesis_points(genesis_id)`.
`genesis_points.anchor_tx_id` references the `txn_id` from `chain_txns`, which is set by `AnchorGenesisPoint`.

During batch execution:
1. `AddSproutsToBatch` is called, which will
- insert into `genesis_points`, associating a `genesis_id` with the genesis outpoint. 
- call `BindMintingBatchWithTx` which sets `asset_minting_batches.genesis_id = genesis_id`
2. In `CommitSignedGenesisTx` 
- `UpsertChainTx` is called, inserting the tx into `chain_txns`, and the `txn_id` is returned.
-  `AnchorGenesisPoint` is called with the `genesisOutpoint` from the tx. This will in turn set `genesis_outpoints.anchor_tx_id = txn_id`.

So if we now join on the `anchor_tx_id = txn_id` we find the `genesis_id` that matches the batch in question, and we should get the correct tx and its height hint.

Reply:
Or we can just store the raw height hint beside the `minting_tx_psbt` in the `asset_minting_batches` table? Seems a lot simpler, and then it can also be used to later on do things like group by that to find all the batches created at a given height.